### DISCUSSION TITLE
How to use dynamic refs in Nuxt 3

### DISCUSSION THREAD

**Discussion by shba007** [nuxt/nuxt#16749] [Questions]
![code](https://user-images.githubusercontent.com/48382105/192351361-18755b2f-a57c-44af-90ed-c2f8607913d3.png)
![code-2](https://user-images.githubusercontent.com/48382105/192351428-4a3580b9-ed55-4a14-bca4-a7bab257e138.png)
![code-3](https://user-images.githubusercontent.com/48382105/192351510-7f4c04fb-99e6-4c12-aeca-f0321e8a9670.png)

console log of home or any other HTMLElement ref is undefined

**Comment by cpreston321** [comment]
@shba007 Can you make reproduction on stackblitz ? I might be able to help with reproduction.

**Reply by shba007** [reply]
https://stackblitz.com/edit/github-clkm1u?file=app.vue

**Comment by TheAlexLichter** [comment]
These refs will be populated only after the elements being mounted (`onMounted` hook). After that (or when using it via `<ClientOnly>`), they also have valid values.

See https://stackblitz.com/edit/https-github-comnuxtframeworkdiscussions7843?file=app.vue,components%2FMyComponent.vue

But the VueUse function seems to throw errors when being used.

PS: Please do not post code as image but use a CodeSandbox/StackBlitz as you did afterwards (best options) *or* post code snippets as text 😋
